test/features/counter/Counter.spec.tsx

```diff
@@ -0,0 +1,161 @@
+  });
+
+  const getWrapper = () =>
+    mount(
+      <Provider store={store}>
+        <Router>
+          <Counter />
+        </Router>
+      </Provider>
+    );
+  const component = getWrapper();
+  return {
+    store,
+    component,
+    actions: counterActions,
+    buttons: component.find('button'),
+    p: component.find('.counter')
+  };
+}
+
+describe('Counter component', () => {
+  it('should should display count', () => {
+    const { p } = setup();
+    expect(p.text()).toMatch(/^1$/);
+  });
+
+  it('should first button should call increment', () => {
+    const { buttons, actions } = setup();
+    const incrementSpy = spy(actions, 'increment');
+
+    buttons.at(0).simulate('click');
+    expect(incrementSpy.called).toBe(true);
+    incrementSpy.restore();
+  });
+
+  it('should match exact snapshot', () => {
+    const { store } = setup();
+    const tree = renderer
+      .create(
+        <Provider store={store}>
+          <Router>
+            <Counter />
+          </Router>
+        </Provider>
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should second button should call decrement', () => {
+    const { buttons, actions } = setup();
+    const decrementSyp = spy(actions, 'decrement');
+    buttons.at(1).simulate('click');
+    expect(decrementSyp.called).toBe(true);
+    decrementSyp.restore();
+  });
+
+  it('should third button should call incrementIfOdd', () => {
+    const { buttons, actions } = setup();
+    const incrementIfOdd = spy(actions, 'incrementIfOdd');
+    buttons.at(2).simulate('click');
+    expect(incrementIfOdd.called).toBe(true);
+    incrementIfOdd.restore();
+  });
+
+  it('should fourth button should call incrementAsync', () => {
+    const { buttons, actions } = setup();
+    const incrementAsync = spy(actions, 'incrementAsync');
+    buttons.at(3).simulate('click');
+    expect(incrementAsync.called).toBe(true);
+    incrementAsync.restore();
+  });
+
+  it('should display updated count after increment button click', () => {
+    const { buttons, p } = setup();
+    buttons.at(0).simulate('click');
+    expect(p.text()).toMatch(/^2$/);
+  });
+
+  it('should display updated count after decrement button click', () => {
+    const { buttons, p } = setup();
+    buttons.at(1).simulate('click');
+    expect(p.text()).toMatch(/^0$/);
+  });
+
+  it('shouldnt change if even and if odd button clicked', () => {
+    const { buttons, p } = setup({ counter: { value: 2 } });
+    buttons.at(2).simulate('click');
+    expect(p.text()).toMatch(/^2$/);
+  });
+
+  it('should change if odd and if odd button clicked', () => {
+    const { buttons, p } = setup({ counter: { value: 1 } });
+    buttons.at(2).simulate('click');
+    expect(p.text()).toMatch(/^2$/);
+  });
+});
+
+describe('Test counter actions', () => {
+  const actions = counterActions;
+  it('should increment should create increment action', () => {
+    expect(actions.increment()).toMatchSnapshot();
```

Comment:
@markerikson I am not sure if these make any sense, test an auto-created action to match with the latest snapshot? :/

Reply:
I don't see any benefit in testing most action creators anyway, and _definitely_ not those that were auto-generated from RTK.  I'd just remove these tests entirely.